xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1566,6 +1567,15 @@ bool CPVRClients::OpenStream(const CPVRRecordingPtr &channel)
       client->OpenStream(channel))
   {
     CSingleLock lock(m_critSection);
+
+    CDateTime endTime = channel->RecordingTimeAsLocalTime() +
+        CDateTimeSpan(0, 0, channel->GetDuration() / 60, channel->GetDuration() % 60);
+
+    m_bIsRecordingInProgress = (endTime > CDateTime::GetCurrentDateTime());
```

Comment:
I'm a bit late to the party, and don't understand this bit of the code, so sorry If I'm stating the obvious or asking something stupid.

I agree with @Glenn-1990. It seems daft to use an (optional) EPG tag reference to determine if a recording is still in progress or not especially if most clients don't implement it yet. The performance penalty is definitely an issue for mythtv too.

If the problem here were simply the localtime dst issue, you could just do everything in UTC.
All PVR API times are passed in UTC anyway (as time_t), so localtime/dst etc... is irrelevant.

However, Duration could be just the 'Duration' of the file recorded so far, so this still might not work reliably as the client probably only updates 'duration' intermittently during a recording. (Duration and recording start time are also optional parameters by the way)

I'm guessing that the objective here is to identify an 'in progress' recording as a growing stream that needs special handling.

Probably it doesn't matter if a stream gets this buffering without needing it, so you could assume that 'Duration' will be refreshed with current recording length at least every 5 minutes. If 'recordingTime' and 'iDuration' aren't kept up to date for a recording then it simply won't get the buffering features.

```
CDateTime endTime = channel->RecordingTimeAsUTC() +
  CDateTimeSpan(0, 0, channel->GetDuration() / 60, channel->GetDuration() % 60) +
  CDateTimeSpan(0, 0, 5, 0);

m_bIsRecordingInProgress = (endTime > CDateTime::GetUTCDateTime());
```
This won't require an API change, or changes to existing pvr code to keep the current behaviour. 

A simple doxy change at https://github.com/xbmc/xbmc/blob/master/xbmc/addons/include/xbmc_pvr_types.h#L441 stating that if iDuration is more than 5 minutes stale during a recording and start time not provided / incorrect that buffering of the stream won't work properly might be enough.

I'm sure there are reasons why this won't work, but it might get someone who knows thinking of something which will.